server/utils/formatter.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * @description Determines the overal status from individual status
+ * @param {Array} automations Array of automations
+ *
+ * @returns {String} success/failure
+ */
+export function getOveralStatus(automations) {
+  if (Array.isArray(automations) && automations.length) {
+    if (automations.every(automation => automation.status === 'success')) {
+      return 'success';
+    }
+    return 'failure';
+  }
+  return 'failure';
+}
+
+/**
+ * @description Formats slack automations data
+ * @param {Array} slackAutomations Array of slack automations
+ *
+ * @returns {Object} Formatted slack automation
+ */
+export function formatSlackAutomations(slackAutomations) {
+  const automations = {};
+
+  automations.status = getOveralStatus(slackAutomations);
+  automations.slackActivities = slackAutomations.map((sa) => {
+    const {
+      status, statusMessage, type, channelId, slackUserId,
+    } = sa;
+    const activity = {
```

Comment:
What if we did:
```
const {'unwanted', 'values', ...activityData} = sa
return activityData;
```
Which means we no longer need new variable: `activity`

Reply:
Done